src/stores/index.js

```diff
@@ -1020,6 +1028,87 @@ class Store {
     })
   }
 
+  getLeaderBoard = (_payload) => {
+    
+    const web3 = new Web3(store.getStore('web3context').library.provider);
+    const memes = this.getMemes();
+
+    async.map(memes, (meme, callback) => {
+        this._getLeaderBoardData(web3, meme, callback);
+    }, async(err, leaderboardData) => {
+      if(err) {
+        return emitter.emit(ERROR, err);
+      }
+      console.log(leaderboardData);
+      leaderboardData.sort(function(a, b) {
+        return b.score - a.score; 
+      })
+  
+      store.setStore({ leaderboard: leaderboardData })
+      emitter.emit(GET_LEADERBOARD_RETURNED);
+      
+    })
+  }
+
+  
+  _getLeaderBoardData = async (web3, meme, callback) => {
+      let element = new Object();
+      element['id'] = meme.id;
+      element['votesFor'] = web3.utils.fromWei(meme.totalForVotes, 'ether');
+      element['votesAgainst'] = web3.utils.fromWei(meme.totalAgainstVotes, 'ether');
+      element['poster'] = meme.proposer;
+      element['voters'] = await getVoterCount(meme.id);
+      let memeScore = 0;
+      let voteAdjustmentFactor = 0;
+
+      if (Number(element['voters']) >= Number(element['votesFor'])) {
+        memeScore = Number(element['voters']) + Number(element['votesFor'])
+      }
+      else {
+        const score = this._calculateMemeScore(Number(element['voters']), Number(element['votesFor']))
+        memeScore = score.score;
+        voteAdjustmentFactor = score.voteAdjustmentFactor;
+      }
+
+      element['score'] = memeScore;
+      element['factor'] = voteAdjustmentFactor;
+
+      const prize = this._calculatePrize(Number(element['voters']), Number(element['votesFor']));
+
+      element['prizeTier'] = prize.totalPrize
+      element['prizeToPoster'] = prize.priceToPoster
+      element['prizePerYFLVote'] = prize.pricePerYFLVote
+      callback(null, element)
```

Comment:
can you please rewrite this in an immutable way?
rather than doing this, you can do something like this.
```
const element = {
  a: ...
  b: ... 
}
```

Reply:
FIXED